(mars theme) aria-current="page" should be retained on menu for eg /blog/page/N/

(sorry if this is not the place for this... should these minor issues just be raised on the forum?)

**_Bug report_**

**Expected behavior**

When visiting eg `/blog/page/2/` the "Blog" item should still be highlighted in the menu. 
this is currently created by
`aria-current={state.router.link === link ? "page" : undefined}`
and so only works on the first page

**Observed behavior**

Blog item is underlined in  menu for the first page, but not for subsequent

**Steps involved to reproduce the problem**

* go to https://5hok7.sse.codesandbox.io/blog/
* click "older posts"

Hello @luisherranz , I can not pass the linter. The following appears to me:

> Warning: React version not specified in eslint-plugin-react settings. See https://github.com/yannickcr/eslint-plugin-react#configuration .
.../frontity/packages/mars-theme/src/components/nav.js
  4:1  error  Missing JSDoc @returns declaration        jsdoc/require-returns
  9:0  error  Missing JSDoc @param "root0" description  jsdoc/require-param-description

--- a/packages/mars-theme-typescript/src/components/nav.tsx
+++ b/packages/mars-theme-typescript/src/components/nav.tsx
@@ -14,7 +14,9 @@ const Nav = (): JSX.Element => {
     <NavContainer>
       {state.theme.menu.map(([name, link]) => {
         // Check if the link matched the current page url.
-        const isCurrentPage = state.router.link === link;
+        const data = state.source.get(state.router.link);
+        const isCurrentPage = data.route === link;
+
         return (
           <NavItem key={name}>
             {/* If link url is the current page, add `aria-current` for a11y */}

--- a/packages/mars-theme/src/components/nav.js
+++ b/packages/mars-theme/src/components/nav.js
@@ -9,12 +9,9 @@ import Link from "./link";
 const Nav = ({ state }) => (
   <NavContainer>
     {state.theme.menu.map(([name, link]) => {
-      // Gets the page number
-      let pageNumber = state.router.link.replace(/\D/g, "");
-
-      // Check if the link matched the current page url, removing page + number from this
-      const isCurrentPage =
-        state.router.link.replace("page/" + pageNumber + "/", "") === link;
+      // Check if the link matched the current page url
+      const data = state.source.get(state.router.link);
+      const isCurrentPage = data.route === link;
 
       return (
         <NavItem key={name}>

--- a/packages/mars-theme/src/components/nav.js
+++ b/packages/mars-theme/src/components/nav.js
@@ -9,8 +9,13 @@ import Link from "./link";
 const Nav = ({ state }) => (
   <NavContainer>
     {state.theme.menu.map(([name, link]) => {
-      // Check if the link matched the current page url
-      const isCurrentPage = state.router.link === link;
+      // Gets the page number
+      let pageNumber = state.router.link.replace(/\D/g, "");
+
+      // Check if the link matched the current page url, removing page + number from this
+      const isCurrentPage =
+        state.router.link.replace("page/" + pageNumber + "/", "") === link;
+
       return (
         <NavItem key={name}>
           {/* If link url is the current page, add `aria-current` for a11y */}

